Expand discussion of resolution order for cloud-provider service auth in doc (#13239)
PrefectHQ/prefect: docs/guides/secrets.md

In Prefect Cloud, these block values are stored in encrypted format.
Organizations whose security policies make such storage infeasible can still use Prefect to connect to third-party services securely.

Any sensitive information that is not stored in a block can be read from the environment.

For example, to find AWS credentials for authentication, any attributes not provided to an AWS Credentials block are sourced at runtime in the order shown in the [Boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials).
Prefect-aws creates the session object using the values in the block and then, any missing values follow the sequence in the Boto3 docs.

Prefect-gcp and prefect-azure follow similar patterns.

> In the example below, we interact with a Snowflake database and store the credentials we need to connect in AWS Secrets Manager.

Reviewer: ```suggestion
In the example below, we interact with a Snowflake database using credentials stored in AWS Secrets Manager.
```

### Create `AwsCredentials` block

Under the hood, Prefect is using the AWS `boto3` client to create a session.

> In the **AwsCredentials** section of the form, click **Add +** and a form will appear to create an AWS Credentials block.

Reviewer: ```suggestion
In the **AwsCredentials** section of the form, click **Add +** and create an AWS Credentials block by entering the necessary values.
```
